kubetest2-tf/pkg/terraform/terraform.go

```diff
@@ -14,68 +15,51 @@ const (
 	StateFileName string = "terraform.tfstate"
 )
 
-func Apply(dir string, platform string, autoApprove bool, extraArgs ...string) (path string, err error) {
+func Apply(dir string, platform string) (path string, err error) {
 	err = unpackAndInit(dir, platform)
 	if err != nil {
-		return "", err
+		return "", fmt.Errorf("Failed to unpack terraform dependencies and Init: %v", err)
 	}
-
-	defaultArgs := []string{
-		"-input=false",
-		fmt.Sprintf("-state=%s", filepath.Join(dir, StateFileName)),
-		fmt.Sprintf("-state-out=%s", filepath.Join(dir, StateFileName)),
+	tf, err := exec.GetTerraformExecutor(dir, platform)
+	if err != nil {
+		return "", err
 	}
-	if autoApprove {
-		defaultArgs = append(defaultArgs, "-auto-approve")
+	if err = tf.Apply(context.Background()); err != nil {
+		return "", fmt.Errorf("failed to apply Terraform: %v", err)
 	}
-	args := append(defaultArgs, extraArgs...)
 	sf := filepath.Join(dir, StateFileName)
-
-	if exitCode := exec.Apply(dir, args); exitCode != 0 {
-		return sf, errors.New("failed to apply Terraform")
-	}
 	return sf, nil
 }
 
-func Destroy(dir string, platform string, autoApprove bool, extraArgs ...string) (err error) {
+func Destroy(dir string, platform string) (err error) {
 	err = unpackAndInit(dir, platform)
 	if err != nil {
-		return err
+		return fmt.Errorf("Failed to unpack terraform dependencies and Init: %v", err)
 	}
-
-	defaultArgs := []string{
-		"-input=false",
-		fmt.Sprintf("-state=%s", filepath.Join(dir, StateFileName)),
-		fmt.Sprintf("-state-out=%s", filepath.Join(dir, StateFileName)),
-	}
-	if autoApprove {
```

Comment:
The module has no support for the `-auto-approve` option and is baked into the module for all terraform operations.
For e.g: [tf.Apply](https://github.com/hashicorp/terraform-exec/blob/96b1e7a5c349c120b7d248c0d28ebf7a517db2fe/tfexec/apply.go#L176)